pkg/util/podstore/podstore_test.go

```diff
@@ -0,0 +1,514 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package podstore
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/wait"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	clock "k8s.io/utils/clock/testing"
+)
+
+var (
+	refTime = time.Now()
+	pod1    = &v1.Pod{
+		Status: v1.PodStatus{
+			PodIPs: []v1.PodIP{
+				{
+					IP: "1.2.3.4",
+				},
+			},
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pod1",
+			Namespace: "pod1_ns",
+			UID:       "pod1",
+		},
+	}
+	pod2 = &v1.Pod{
+		Status: v1.PodStatus{
+			PodIPs: []v1.PodIP{
+				{
+					IP: "5.6.7.8",
+				},
+			},
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pod2",
+			Namespace: "pod2_ns",
+			UID:       "pod2",
+		},
+	}
+	timestampMap = map[string]*LocalClock{
+		"pod1": {CreationTimestamp: refTime},
+		"pod2": {CreationTimestamp: refTime.Add(-5 * time.Minute), DeletionTimestamp: &refTime},
+	}
+	node = &v1.Node{}
+)
+
+func Test_onPodCreate(t *testing.T) {
+	podStore := &PodStore{
+		timestampMap: map[string]*LocalClock{},
+		clock:        clock.NewFakeClock(refTime),
+	}
+	tests := []struct {
+		name                    string
+		podStore                *PodStore
+		obj                     interface{}
+		expectedTimestampMapLen int
+	}{
+		{
+			name:                    "object is not Pod",
+			podStore:                podStore,
+			obj:                     node,
+			expectedTimestampMapLen: 0,
+		},
+		{
+			name:                    "Valid case",
+			obj:                     pod1,
+			podStore:                podStore,
+			expectedTimestampMapLen: 1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equal(t, 0, len(tt.podStore.timestampMap))
+			podStore.onPodCreate(tt.obj)
+			assert.Equal(t, tt.expectedTimestampMapLen, len(tt.podStore.timestampMap))
+			if tt.expectedTimestampMapLen > 0 {
+				assert.Equal(t, tt.podStore.clock.Now(), tt.podStore.timestampMap[string(tt.obj.(*v1.Pod).UID)].CreationTimestamp)
+			}
+		})
+	}
+}
+
+func Test_onPodDelete(t *testing.T) {
+	fakeClock := clock.NewFakeClock(refTime)
+	podStore := PodStore{
+		curPod:       nil,
+		prevPod:      cache.NewIndexer(podKeyFunc, cache.Indexers{PodIPIndex: podIPIndexFunc}),
+		podsToDelete: workqueue.NewDelayingQueueWithCustomClock(fakeClock, DeleteQueueName),
+		timestampMap: map[string]*LocalClock{"pod1": {CreationTimestamp: refTime}},
+		clock:        clock.NewFakeClock(refTime),
+	}
```

Comment:
IMO, it's bad practice to reuse the same clock and Pod store for multiple test cases, as there could be some interference

Reply:
Fixed. Thanks